backend/shork/src/main/kotlin/internal/memory/ICore.kt

```diff
@@ -11,9 +10,5 @@ internal interface ICore {
     /** Writes an instruction to the given address */
     fun storeAbsolute(address: Int, instruction: AbstractInstruction)
 
-    /** Resolves the address of a field based on the address mode and the content of the field */
-    fun resolveForReading(sourceAddress: Int, field: Int, mode: AddressMode): Int
-
-    /** Resolves the address of a field based on the address mode and the content of the field */
-    fun resolveForWriting(sourceAddress: Int, field: Int, mode: AddressMode): Int
+    fun resolveAll(sourceAddress: Int): ResolvedAddresses
```

Comment:
A doc comment like the ones above had would be nice.
You could maybe also rename it to something like `resolveFields` or `resolveAddresses` or smth

Reply:
Renamed to resolveFields and added doc comment